writers/writers_test.go

```diff
@@ -0,0 +1,221 @@
+	}
+	const multiplyN = 1000
+
+	var bCases []bCase
+	bCases = append(bCases, writerMatrix("BatchWriter", batchwriter.New, newBatchWriterClient(), makeRecord, batchwriterOpts)...)
+	bCases = append(bCases, writerMatrix("BatchWriter wide", batchwriter.New, newBatchWriterClient(), makeWideRecord, batchwriterOpts)...)
+	bCases = append(bCases, writerMatrix("MixedBatchWriter", mixedbatchwriter.New, newMixedBatchWriterClient(), makeRecord, mixedbatchwriterOpts)...)
+	bCases = append(bCases, writerMatrix("MixedBatchWriter wide", mixedbatchwriter.New, newMixedBatchWriterClient(), makeWideRecord, mixedbatchwriterOpts)...)
+	bCases = append(bCases, writerMatrix("StreamingBatchWriter", streamingbatchwriter.New, newStreamingBatchWriterClient(), makeRecord, streamingbatchwriterOpts)...)
+	bCases = append(bCases, writerMatrix("StreamingBatchWriter wide", streamingbatchwriter.New, newStreamingBatchWriterClient(), makeWideRecord, streamingbatchwriterOpts)...)
+
+	for _, c := range bCases {
+		c := c
+		b.Run(c.name, func(b *testing.B) {
+			var (
+				mStart runtime.MemStats
+				mEnd   runtime.MemStats
+			)
+
+			ch := make(chan message.WriteMessage)
+			errCh := make(chan error)
+			go func() {
+				defer close(errCh)
+				errCh <- c.wr.Write(context.Background(), ch)
+			}()
+
+			newN := b.N * multiplyN
+
+			runtime.ReadMemStats(&mStart)
+			b.ResetTimer()
+			for i := 0; i < newN; i++ {
+				rec := c.rec()
+				ch <- &message.WriteInsert{
+					Record: rec,
+				}
+			}
+			close(ch)
+			err := <-errCh
+
+			b.StopTimer()
+
+			if err != nil {
+				b.Fatal(err)
+			}
+
+			runtime.ReadMemStats(&mEnd)
+
+			allocatedBytes := mEnd.Alloc - mStart.Alloc
+			b.ReportMetric(float64(allocatedBytes)/float64(newN), "bytes/op")
+		})
+	}
+}
+
+func makeRecord() func() arrow.Record {
+	table := &schema.Table{
+		Name: "test_table",
+		Columns: schema.ColumnList{
+			{
+				Name: "col1",
+				Type: arrow.BinaryTypes.String,
+			},
+		},
+	}
+	sc := table.ToArrowSchema()
+
+	return func() arrow.Record {
+		bldr := array.NewRecordBuilder(memory.DefaultAllocator, sc)
+		bldr.Field(0).(*array.StringBuilder).Append("test")
+		return bldr.NewRecord()
+	}
+}
+
+func makeWideRecord() func() arrow.Record {
+	table := &schema.Table{
+		Name: "test_wide_table",
+		Columns: schema.ColumnList{
+			{
+				Name: "col1",
+				Type: arrow.BinaryTypes.String,
+			},
+		},
+	}
+
+	const numWideCols = 200
+	randVals := make([]int64, numWideCols)
+	for i := 0; i < numWideCols; i++ {
+		table.Columns = append(table.Columns, schema.Column{
+			Name: "wide_col" + strconv.Itoa(i),
+			Type: arrow.PrimitiveTypes.Int64,
+		})
+		randVals[i] = rand.Int63()
+	}
+	sc := table.ToArrowSchema()
+
+	return func() arrow.Record {
+		bldr := array.NewRecordBuilder(memory.DefaultAllocator, sc)
+		bldr.Field(0).(*array.StringBuilder).Append("test")
+		for i := 0; i < numWideCols; i++ {
+			bldr.Field(i + 1).(*array.Int64Builder).Append(randVals[i])
```

Comment:
```suggestion
		for i := 1; i <= numWideCols; i++ {
			bldr.Field(i).(*array.Int64Builder).Append(randVals[i])
```

Reply:
you'd need to do `randVals[i-1]`, adds compexity and potential copy/paste errors, just to avoid an `+ 1`